kafka-rest/src/test/java/io/confluent/kafkarest/resources/v3/ClusterConfigResourceTest.java

```diff
@@ -0,0 +1,323 @@
+
+    assertEquals(expected, response.getValue());
+  }
+
+  @Test
+  public void getClusterConfig_nonExistingConfig_throwsNotFound() {
+    expect(clusterConfigManager.getClusterConfig(
+        CLUSTER_ID, ClusterConfig.Type.BROKER, CONFIG_1.getName()))
+        .andReturn(failedFuture(new NotFoundException()));
+    replay(clusterConfigManager);
+
+    FakeAsyncResponse response = new FakeAsyncResponse();
+    clusterConfigsResource.getClusterConfig(
+        response, CLUSTER_ID, ClusterConfig.Type.BROKER, CONFIG_1.getName());
+
+    assertEquals(NotFoundException.class, response.getException().getClass());
+  }
+
+  @Test
+  public void getClusterConfig_nonExistingCluster_throwsNotFound() {
+    expect(clusterConfigManager.getClusterConfig(
+        CLUSTER_ID, ClusterConfig.Type.BROKER, CONFIG_1.getName()))
+        .andReturn(failedFuture(new NotFoundException()));
+    replay(clusterConfigManager);
+
+    FakeAsyncResponse response = new FakeAsyncResponse();
+    clusterConfigsResource.getClusterConfig(
+        response, CLUSTER_ID, ClusterConfig.Type.BROKER, CONFIG_1.getName());
+
+    assertEquals(NotFoundException.class, response.getException().getClass());
+  }
+
+  @Test
+  public void updateClusterConfig_existingConfig_updatesConfig() {
+    expect(
+        clusterConfigManager.upsertClusterConfig(
+            CLUSTER_ID,
+            ClusterConfig.Type.BROKER,
+            CONFIG_1.getName(),
+            "new-value"))
+        .andReturn(completedFuture(null));
+    replay(clusterConfigManager);
+
+    FakeAsyncResponse response = new FakeAsyncResponse();
+    clusterConfigsResource.upsertClusterConfig(
+        response,
+        CLUSTER_ID,
+        ClusterConfig.Type.BROKER,
+        CONFIG_1.getName(),
+        new UpdateClusterConfigRequest(
+            new UpdateClusterConfigRequest.Data(
+                new UpdateClusterConfigRequest.Data.Attributes("new-value"))));
+    assertNull(response.getValue());
+    assertNull(response.getException());
+    assertTrue(response.isDone());
+  }
+
+  @Test
+  public void updateConfig_nonExistingCluster_throwsNotFound() {
+    expect(
+        clusterConfigManager.upsertClusterConfig(
+            CLUSTER_ID,
+            ClusterConfig.Type.BROKER,
+            CONFIG_1.getName(),
+            "new-value"))
+        .andReturn(failedFuture(new NotFoundException()));
+    replay(clusterConfigManager);
+
+    FakeAsyncResponse response = new FakeAsyncResponse();
+    clusterConfigsResource.upsertClusterConfig(
+        response,
+        CLUSTER_ID,
+        ClusterConfig.Type.BROKER,
+        CONFIG_1.getName(),
+        new UpdateClusterConfigRequest(
+            new UpdateClusterConfigRequest.Data(
+                new UpdateClusterConfigRequest.Data.Attributes("new-value"))));
+
+    assertEquals(NotFoundException.class, response.getException().getClass());
+  }
+
+  @Test
+  public void resetClusterConfig_existingConfig_resetsConfig() {
```

Comment:
Test for resetting for nonExistingConfig.

Reply:
As explained in the other comment, we can't fail if the config does not exist. It's kind of a lame behaviour which if I have time I will try to fix in AK at some point.